skyportal/handlers/api/spectrum.py

```diff
@@ -80,17 +93,10 @@ def post(self):
         single_user_group = self.associated_user_object.single_user_group
 
         group_ids = data.pop("group_ids", None)
-        if group_ids is None:
+        if group_ids == [] or group_ids is None:
             groups = [single_user_group]
         else:
-            if group_ids == "all":
-                groups = Group.query.filter(
-                    Group.name == cfg['misc.public_group_name']
-                ).all()
-            else:
-                groups = Group.get_if_accessible_by(
-                    group_ids, self.current_user, raise_if_none=True
-                )
+            groups = self.get_groups(group_ids)
 
         if single_user_group not in groups:
```

Comment:
if groups can possibly be None than this will raise an exception since None is not iterable (but I think the logic above is that groups cannot be None here.

Reply:
Yes, I believe the logic is that if group_ids is None then the groups variable will be a list with only the single user group.